src/common/low_precision_transformations/src/recurrent_cell.cpp

```diff
@@ -0,0 +1,176 @@
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset1.hpp>
+
+#include <memory>
+#include <ngraph/node.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/pattern/op/or.hpp>
+
+#include "low_precision/network_helper.hpp"
+#include "../include/low_precision/rt_info/skip_cleanup_attribute.hpp"
+
+namespace ngraph {
+namespace pass {
+namespace low_precision {
+
+RecurrentCellTransformation::RecurrentCellTransformation(const Params& params) : LayerTransformation(params) {
+    const auto X = ngraph::pattern::wrap_type<ngraph::opset1::Parameter>();
+    const auto H = ngraph::pattern::wrap_type<ngraph::opset1::Parameter>();
+    const auto C = ngraph::pattern::wrap_type<ngraph::opset1::Parameter>();
+    const auto W = ngraph::pattern::wrap_type<ngraph::opset1::Constant>();
+    const auto R = ngraph::pattern::wrap_type<ngraph::opset1::Constant>();
+    const auto B = ngraph::pattern::wrap_type<ngraph::opset1::Constant>();
+
+    const auto fq_X = NetworkHelper::fakeQuantizeWraper(X);
+    const auto fq_H = NetworkHelper::fakeQuantizeWraper(H);
+    const auto fq_W = NetworkHelper::fakeQuantizeWraper(W);
+    const auto fq_R = NetworkHelper::fakeQuantizeWraper(R);
+
+    const auto dequantization_convert_X = ngraph::pattern::wrap_type<ngraph::opset1::Convert>({ngraph::pattern::any_input()});
+    const auto dequantization_convert_H = ngraph::pattern::wrap_type<ngraph::opset1::Convert>({ngraph::pattern::any_input()});
+    const auto subtract_constant = ngraph::pattern::wrap_type<ngraph::opset1::Constant>();
+    const auto dequantization_subtract_X = ngraph::pattern::wrap_type<ngraph::opset1::Subtract>(
+        {dequantization_convert_X, subtract_constant});
+    const auto dequantization_subtract_H = ngraph::pattern::wrap_type<ngraph::opset1::Subtract>(
+        {dequantization_convert_H, subtract_constant});
+    const auto multiply_constant = ngraph::pattern::wrap_type<ngraph::opset1::Constant>();
+    const auto dequantization_multiply_X = ngraph::pattern::wrap_type<ngraph::opset1::Multiply>(
+        {dequantization_subtract_X, multiply_constant});
+
+    const auto dequantization_multiply_without_subtract_X = ngraph::pattern::wrap_type<ngraph::opset1::Multiply>(
+        {dequantization_convert_X, multiply_constant});
+    const auto dequantization_multiply_H = ngraph::pattern::wrap_type<ngraph::opset1::Multiply>(
+        {dequantization_subtract_H, multiply_constant});
+    const auto dequantization_multiply_without_subtract_H = ngraph::pattern::wrap_type<ngraph::opset1::Multiply>(
+        {dequantization_convert_H, multiply_constant});
+
+    const auto lstm_cell = ngraph::pattern::wrap_type<ngraph::opset5::LSTMCell>(
+        {fq_X, fq_H, C, fq_W, fq_R, B});
+    const auto lstm_cell_with_dequantizations = ngraph::pattern::wrap_type<ngraph::opset5::LSTMCell>(
+        {dequantization_multiply_X, dequantization_multiply_H, C, fq_W, fq_R, B});
```

Comment:
Refactoring idea: not clear source code with copy/paste. Can we refactor wrapper creation for dequantization patter - extract to new function `wrap_dequantization`, please? Like as:
``` cpp
const auto lstm_cell_with_dequantizations = ngraph::pattern::wrap_type<ngraph::opset5::LSTMCell>({wrap_dequantization(X), wrap_dequantization(H), C, wrap_fake_quantize(W), wrap_fake_quantize(R), B});
```

Reply:
done, thank you. Also added wrap_quantization.